rel/m_lib.c

```diff
@@ -1,371 +1,817 @@
+  if ((-min <= x) && (x <= min)) {
+    return 0.0f;
+  }
+  if (x >= max) {
+    return 1.0f;
+  }
+  if (x <= -max) {
+    return -1.0f;
+  }
+  if (shift_by_min) {
+    if (x > 0.0f) {
+      return (x - min) * scale;
+    } else {
+      return (x + min) * scale;
+    }
+  } else {
+    return x * scale;
+  }
 }
 
-// f32 check_percent_abs(s32 arg0, f32 arg8, f32 arg9, f32 argA, f32 argB)
-// f32 get_percent_forAccelBrake(f32 arg8, f32 arg9, f32 argA, f32 argB, f32 argC)
-
-
-void Game_play_Projection_Trans(game_play* play, Vec3f* src, Vec3f* dest) {
-    f32 w;
+/**
+ * @brief Calculate the percentage of completion based on acceleration and braking values.
+ *
+ * This function calculates the percentage of completion using the current position, start and end positions,
+ * acceleration and braking distances using quadratic easing.
+ *
+ * @param now Current position value.
+ * @param start Start position value.
+ * @param end End position value.
+ * @param accelerate Acceleration distance.
+ * @param brake Braking distance.
+ * @return Percentage of completion.
+ */
+extern f32 get_percent_forAccelBrake(f32 now, f32 start, f32 end,
+                                     f32 accelerate, f32 brake) {
+  f32 percent;
+  f32 total_delta;
+  f32 now_delta;
+  f32 step;
+
+  if (now >= end) {
+    return 1.0f;
+  }
+  if (now <= start) {
+    return 0.0f;
+  }
+  total_delta = end - start;
+  now_delta = now - start;
+  if (total_delta < (accelerate + brake)) {
+    return 0.0f;
+  }
+
+  step = 1.0f / (((2.0f * total_delta) - accelerate) - brake);
+  if (accelerate != 0.0f) {
+    if (now_delta <= accelerate) {
+      percent = (now_delta * (step * now_delta));
+      percent /= accelerate;
+      return percent;
+    }
+    percent = step * accelerate;
+  } else {
+    percent = 0.0f;
+  }
+  if (now_delta <= (total_delta - brake)) {
+    percent += (step * 2.0f) * (now_delta - accelerate);
+    return percent;
+  }
+  percent += (2.0f * step * ((total_delta - accelerate) - brake));
+  if (brake != 0.0f) {
+    percent += step * brake;
+    if (now_delta < total_delta) {
+      f32 diff = total_delta - now_delta;
+      percent -= step * diff * diff / brake;
+    }
+  }
+  return percent;
+}
 
-    Matrix_mult(&play->unk_200C, 0);
-    Matrix_Position(src, dest);
-    w = play->unk_200C.ww + (
-          (play->unk_200C.wx * src->x) 
-        + (play->unk_200C.wy * src->y) 
-        + (play->unk_200C.wz * src->z));
-    dest->x = 160.0f + ((dest->x / w) * 160.0f);
-    dest->y = 120.0f - ((dest->y / w) * 120.0f);
+/**
+ * @brief Project a 3D world position onto a 2D screen position.
+ *
+ * @param play Pointer to a GAME_PLAY structure.
+ * @param wpos Pointer to the 3D world position (xyz_t).
+ * @param screen_pos Pointer to the resulting 2D screen position (xyz_t).
+ */
+extern void Game_play_Projection_Trans(GAME_PLAY* play, xyz_t* wpos,
+                                       xyz_t* screen_pos) {
```

Comment:
Possibly it will turn out that screen positions can use a different typedef for clarity.